code/modules/power/tesla/energy_ball.dm

```diff
@@ -79,9 +79,12 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
 	var/first_move = dir
 	for(var/i in 0 to move_amount)
 		var/move_dir = pick(alldirs + first_move) //give the first move direction a bit of favoring.
+		if(target && prob(60))
```

Comment:
You need to make sure target is on the same zlevel as the tesla.

just do `if(target && target.z = z && prob(60))`